app/src/main/java/com/github/se/icebreakrr/model/profile/ProfilesViewModel.kt

```diff
@@ -378,6 +382,42 @@ open class ProfilesViewModel(
         onFailure = { e -> handleError(e) })
   }
 
+  /**
+   * Adds user to current user's already met list in the repository.
+   *
+   * @param uid The unique ID of the user being added.
+   */
+  fun addAlreadyMet(uid: String) {
+    _selfProfile.update { currentProfile ->
+      currentProfile?.copy(hasAlreadyMet = currentProfile.hasAlreadyMet + uid)
+    }
+    updateProfile(selfProfile.value!!)
+  }
+
+  /**
+   * Removes user to current user's already met list in the repository.
+   *
+   * @param uid The unique ID of the user being removed.
+   */
+  fun removeAlreadyMet(uid: String) {
+    _selfProfile.update { currentProfile ->
+      currentProfile?.copy(hasAlreadyMet = currentProfile.hasAlreadyMet.filter { it != uid })
+    }
+    updateProfile(selfProfile.value!!)
+    getBlockedUsers()
+  }
+
```

Comment:
You could make one update function for addAlreadyMet and removeAlreadyMet with something like : 

```
fun updateAlreadyMet(uid: String, add: Boolean) {
    _selfProfile.update { currentProfile ->
        currentProfile?.copy(
            hasAlreadyMet = if (add) {
                currentProfile.hasAlreadyMet + uid
            } else {
                currentProfile.hasAlreadyMet.filter { it != uid }
            }
        )
    }
} 
```

But I guess having both part separated can help us during testing and maintaining a clean code.